dbms/src/AggregateFunctions/AggregateFunctionQuantile.cpp

```diff
@@ -52,10 +49,10 @@ static constexpr bool SupportDecimal()
 }
 
 
-template <template <typename> class Function, bool have_second_arg>
+template <template <typename> class Function>
 AggregateFunctionPtr createAggregateFunctionQuantile(const std::string & name, const DataTypes & argument_types, const Array & params)
 {
-    if constexpr (have_second_arg)
+    if constexpr (Function<void>::hasSecondArg()) // FIXME: a little bit hacky to specify concrete type to call a static method.
```

Comment:
It would be clearer to extract entire condition to function. Something like that:
constexpr void Function::assertOnUnexpected(name, argument_types)
{
    if constexpr (has_second_arg)
        assertBinary(...);
    else
        assertUnary(...);
}